### PR TITLE
fix: update tests for subset of two sets challenge

### LINKED ISSUE #17815: Passes without changing code Perform a Subset Check on Two Sets of Data
#### Describe your problem and - if possible - how to reproduce it
This challenge can be passed without changing the code in the editor at all. Just click "Run the Test" and it passes.

#### Add a Link to the page with the problem
https://learn.freecodecamp.org/coding-interview-prep/data-structures/perform-a-subset-check-on-two-sets-of-data/

#### Tell us about your browser and operating system

* Browser Name: Chrome
* Browser Version: Version 67.0.3396.99 (Official Build) (64-bit)
* Operating System: Ubuntu Linux 16.04

### PR DESCRIPTION
**Pull request by cazyw**
For Data Structures: Perform a Subset Check on Two Sets of Data:
Updates the test assertions as currently the tests pass without any code changes

- [X] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] None of my changes are plagiarized from another source without proper attribution.
- [X] My article does not contain shortened URLs or affiliate links.

Closes #17815

### CODE REVIEW

**Comment by cazyw** [comment]
Is also related to #17856

**Comment by cazyw** [comment]
Updated to also include the non-English tests

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/perform-a-subset-check-on-two-sets-of-data.english.md [hunk 1]
```diff
@@ -21,17 +21,17 @@ For example, if <code>setA = ['a','b']</code> and <code>setB = ['a','b','c','d']
...
+    testString: 'assert((function(){var setA = new Set(); var setB = new Set(); setA.add("a"); setA.add("b"); setB.add("a"); setB.add("b"); setB.add("c"); setB.add("d"); var subsetSetAB = setA.subset(setB); return (subsetSetAB === true)})(), "<code>["a", "b"].subset(["a", "b", "c", "d"])</code> should return <code>true</code>");'
   - text: <code>['a', 'b', 'c'].subset(['a', 'b'])</code> should return <code>false</code>")
-    testString: assert(function(){var setA = new Set(); var setB = new Set(); setA.add('a'); setA.add('b'); setA.add('c'); setB.add('a'); setB.add('b'); var subsetSetAB = setA.subset(setB); return (subsetSetAB === false)}, "<code>['a', 'b', 'c'].subset(['a', 'b'])</code> should return <code>false</code>");
+    testString: 'assert((function(){var setA = new Set(); var setB = new Set(); setA.add("a"); setA.add("b"); setA.add("c"); setB.add("a"); setB.add("b"); var subsetSetAB = setA.subset(setB); return (subsetSetAB === false)})(), "<code>["a", "b", "c"].subset(["a", "b"])</code> should return <code>false</code>");'
```
Comment (tbushman) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/perform-a-subset-check-on-two-sets-of-data.english.md]: we need to either escape the double quotes or use single

Comment (tbushman) [reply curriculum/challenges/english/08-coding-interview-prep/data-structures/perform-a-subset-check-on-two-sets-of-data.english.md]: ? Not sure but this is the file with errors, `missing ) after argument list`

**Comment by nikrb** [comment]
@RandellDawson  @scissorsneedfoodtoo  @tbushman  successfully tested english version locally. Propagated `testString` changes to other languages. Can someone else test it locally please?

@cazyw  The brackets were causing issues, so I've rewitten the texty bits trying to follow the [challenge guide](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/how-to-work-on-coding-challenges.md#challenge-template), I hope that's ok. It's only my take so please feel free to update as you see fit.

**Comment by cazyw** [comment]
yep, thanks @nikrb , happy for any changes needed to be made

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/perform-a-subset-check-on-two-sets-of-data.english.md (hunk 1 above)
Comment (nikrb) [reply curriculum/challenges/english/08-coding-interview-prep/data-structures/perform-a-subset-check-on-two-sets-of-data.english.md]: fixed missing bracket

**Comment by RandellDawson** [comment]
@cazyw I noticed you have another PR #18698 which also has changes for the same sections of some of the same files.  This is  going to cause a major merge conflict. I will have to review further to see how we can minimize this issue.

**Comment by cazyw** [comment]
Ah you're right. Sorry I think in PR #18698 I accidentally added in the `perform-a-subset-check-on-two-sets-of-data.english.md` file and then tried to remove it.

**Comment by QuincyLarson** [comment]
@cazyw First of all, thank you for going to the effort of fixing these tests and applying your changes to the other languages as well. 

We are planning tools to help the maintainers of other languages sync their tests and seed code with the other language versions, so that changes to one language (such as your improvements to the English language version) can propagate to other languages, too.

In this case, it sounds like these changes you've made are ready to be merged. Instead of merging this PR, could you open a new PR with just the changes to the English language files? Then we can merge that. I will go ahead and close this current PR.

Then once we've finished our tool, we'll be able to highlight these differences and the maintainers of the other languages will be able to look at them and decide whether to merge them into those languages as well.

**Comment by cazyw** [comment]
ok thanks, will do

**Pull request closed without merging**